plugins/scaffolder/src/components/fields/OwnershipEntityRefPicker/OwnershipEntityRefPicker.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { identityApiRef, useApi } from '@backstage/core-plugin-api';
+import { TextField, FormControl } from '@material-ui/core';
+import {
+  OwnershipEntityRefPickerProps,
+  OwnershipEntityRefPickerSchema,
+} from './schema';
+import { Autocomplete } from '@material-ui/lab';
+import { catalogApiRef } from '@backstage/plugin-catalog-react';
+import { NotFoundError } from '@backstage/errors';
+
+export { OwnershipEntityRefPickerSchema };
+
+export const OwnershipEntityRefPicker = (
+  props: OwnershipEntityRefPickerProps,
+) => {
+  const {
+    schema: { title, description },
+    required,
+    rawErrors,
+  } = props;
+
+  const identityApi = useApi(identityApiRef);
+  const catalogApi = useApi(catalogApiRef);
+  const [groups, setGroups] = useState<string[]>([]);
+  const [selectedGroup, setSelectedGroup] = useState<string | null>(null);
+
+  useEffect(() => {
```

Comment:
Can you not make this `useEffect(async() => {..})` or use `useAsync` instead of having to wrap up the `fetchUserGroups`?

Reply:
Hey @benjdlambert 👋 !

Happy to look into this and try to make this change, would it be okay if you help me understand why that would be better and maybe a small example? I don't know all the ins and outs of react hooks 😅  so just wondering why this is not an ideal approach

Thanks a lot!

Reply:
It's just a little cleaner, and it will do the caching for you. It's much easier to read, and you have better control over the async promise that it's wrapped up with.

For instance you can do something like this:


```ts
const { data, loading, error } = useAsync( () => ...)
```

And you will get the return value, the loading state if the promise is not resolved yet, and the any error thats thrown from the promise, which is a nice pattern even if they're not used so much it's cleaner than using `useEffect` :)